### PR TITLE
Add check-cla comment to force rechecking of cncf cla.

### PR DESCRIPTION
**Pull request by amwat**
Commenting "/check-cla" will check for commit statuses and add the necessary label derived from "cla/linuxfoundation" status events.

Fixes [#6939](https://github.com/kubernetes/test-infra/issues/6939)

### CODE REVIEW

**Comment by BenTheElder** [comment]
@amwat quick guess is that your commits aren't signed with the right email since they also don't appear to be associated with your account.

File: prow/plugins/cla/cla.go [hunk 1]
```diff
@@ -174,3 +182,79 @@ func handle(gc gitHubClient, log *logrus.Entry, se github.StatusEvent) error {
...
+	return handleComment(pc.GitHubClient, pc.Logger, &ce)
+}
+
+func handleComment(gc gitHubClient, log *logrus.Entry, e *github.ReviewCommentEvent) error {
+	// Only consider open PRs and new comments.
+	if e.PullRequest.State != "open" || e.Action != github.ReviewCommentActionCreated {
+		return nil
+	}
+	// If we create a "/check-cla comment, check for the cla in commit statuses, add cla label if necessary.
+	checkCLA := false
```
Comment (fejta) [inline prow/plugins/cla/cla.go]: This seems unnecessary (either it is true or we return).

File: prow/plugins/cla/cla.go [hunk 2]
```diff
@@ -174,3 +182,79 @@ func handle(gc gitHubClient, log *logrus.Entry, se github.StatusEvent) error {
...
+	repo := e.Repo.Name
+	number := e.PullRequest.Number
+	hasCLAYes := false
+	hasCLANo := false
+
+	labels, err := gc.GetIssueLabels(org, repo, number)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get the labels on %s/%s#%d.", org, repo, number)
+	}
+	for _, candidate := range labels {
```
Comment (fejta) [inline prow/plugins/cla/cla.go]: Seems like for both this loop and the `range statuses` loop we should consider breaking out early and/or better handling the edge cases

File: prow/plugins/cla/cla.go (hunk 1 above)
Comment (amwat) [reply prow/plugins/cla/cla.go]: done.

File: prow/plugins/cla/cla.go (hunk 2 above)
Comment (amwat) [reply prow/plugins/cla/cla.go]: We could break out if we are absolutely sure that both flags can't simultaneously exist (for some reason).

File: prow/plugins/cla/cla.go [hunk 3]
```diff
@@ -54,8 +55,13 @@ It may take a couple minutes for the CLA signature to be fully registered; after
 	maxRetries = 5
 )
 
+var (
+	checkCLARe = regexp.MustCompile(`(?mi)^/check-cla\s*$`)
+)
+
 func init() {
 	plugins.RegisterStatusEventHandler(pluginName, handleStatusEvent, helpProvider)
+	plugins.RegisterReviewCommentEventHandler(pluginName, handleCommentEvent, helpProvider)
```
Comment (fejta) [inline prow/plugins/cla/cla.go]: Use `RegisterGenericCommentHandler` -- I should be able to activate this plugin with a comment on either a review comment or an issue comment.


https://github.com/kubernetes/test-infra/blob/master/prow/plugins/plugins.go#L114

File: prow/plugins/cla/cla_test.go [hunk 4]
```diff
@@ -170,3 +170,207 @@ func TestCLALabels(t *testing.T) {
...
+			action:  "created",
+			body:    "/check-cla",
+			pullRequest: github.PullRequest{
+				State:  "open",
+				Number: 3,
+				Head: github.PullRequestBranch{
+					SHA: "sha",
+				},
+			},
+			hasCLAYes: true,
```
Comment (fejta) [inline prow/plugins/cla/cla_test.go]: Add a scenario for `hasCLAYes && hasCLANo`?

**Comment by BenTheElder** [comment]
@amwat can you fix up the commits and let's get this merged?

File: prow/plugins/cla/cla_test.go [hunk 5]
```diff
@@ -170,3 +170,219 @@ func TestCLALabels(t *testing.T) {
...
+			pullRequests: []github.PullRequest{
+				{Number: 3, Head: github.PullRequestBranch{SHA: "sha"}},
+			},
+			hasCLANo:  true,
+			hasCLAYes: true,
+
+			removedLabel: fmt.Sprintf("/#3:%s", claYesLabel),
+		},
+	}
+	for _, tc := range testcases {
```
Comment (fejta) [inline prow/plugins/cla/cla_test.go]: Consider taking advantage of `t.Run(tc.name, func() { ... })` (I'd like us to start using this pattern, where possible)

https://golang.org/pkg/testing/#T.Run

**Comment by fejta** [comment]
/lgtm
/hold

Give a `/hold cancel` when you're ready for this to go in

**Pull request merged**